modules/cxf/src/test/java/org/mule/module/cxf/CxfInboundMessageProcessorTestCase.java

```diff
@@ -105,4 +126,77 @@ private CxfInboundMessageProcessor createCxfMessageProcessor() throws MuleExcept
         return processor;
     }
 
+    @Test
+    public void testInboundWithValidationEnabled() throws Exception
+    {
+        CxfInboundMessageProcessor processor = createMessageProcessorWithValidationEnabled();
+
+        MessageProcessor messageProcessor = new MessageProcessor()
+        {
+            public MuleEvent process(MuleEvent event) throws MuleException
+            {
+                payload = event.getMessage().getPayload();
+                assertThat("echo", equalTo(payload));
+                event.getMessage().setPayload("echo");
```

Comment:
why are you setting the same payload here?
You can set a different value and use that to assert instead of the `gotEvent` flag.